### PR TITLE
add Thai Language translation support

### PR DESCRIPTION
**Pull request by Rinechxn**
# Add Thai Language Translation Support

This PR introduces support for the Thai language across various components in the application. It includes translations for UI messages, dialogs, modals, and notifications to enhance the user experience for Thai-speaking users.

### Testing:
- Verified that the translations display correctly within the UI.
- Ensured that fallback mechanisms work for untranslated strings.

### Co-Maintained By:
- GaraGuide (@gara-guide.bsky.social)

### Screenshots:
image

### CODE REVIEW

**Comment by surfdude29** [comment]
This is great to see, nice work! Especially since some recent language statistics showed that [Thai surged into 5th place](https://bsky.app/profile/surfdude29.ispost.ing/post/3l6wlwpmdp32x) in the table of most popular languages used in posts.

Would it be OK to open a PR in the repo to add some extra changes that are needed to enable the Thai localization?

**Comment by Rinechxn** [comment]
Yes, i want to help and support to thai community

**Comment by surfdude29** [comment]
> Would it be OK to open a PR in the repo to add some extra changes that are needed to enable the Thai localization?

I hope it's OK, I went ahead and opened a PR to add the necessary changes to enable the Thai localization in the app: https://github.com/Rinechxn/social-app/pull/1

File: src/locale/languages.ts [hunk 1]
```diff
@@ -46,6 +47,7 @@ export const APP_LANGUAGES: AppLanguageConfig[] = [
   {code2: AppLanguage.ko, name: '한국어 – Korean'},
   {code2: AppLanguage.pt_BR, name: 'Português (BR) – Portuguese (BR)'},
   {code2: AppLanguage.ru, name: 'Русский – Russian'},
+  {code2: AppLanguage.th, name: 'ภาษาไทย – Thai'},
```
[the remark this answers is not in the document]
Comment (auroursa) [reply src/locale/languages.ts:50]: Perhaps using “อักษรไทย” would be more appropriate here? “อักษรไทย” represents the written form, and “ภาษาไทย” refers to a spoken form, there is a subtle difference between there.

[the remark this answers is not in the document]
Comment (surfdude29) [reply src/locale/languages.ts:50]: Ah, good catch @auroursa 👍  I hadn't realised there was a difference I have to admit, I copied and pasted what was in the infobox for the Thai language article on Wikipedia 😄

@Rinechxn will know what is the best term to use :)

[the remark this answers is not in the document]
Comment (Rinechxn) [reply src/locale/languages.ts:50]: Personally, I prefer the term "ภาษาไทย" (Thai language) as it feels more formal and easier to say.

**Comment by pfrazee** [comment]
@Rinechxn Thank you so much! @auroursa are you a Thai speaker? If so, was your review a signoff? If not, @Rinechxn  is there somebody we could recruit to act as a second pair of eyes on this?

**Comment by surfdude29** [comment]
@pfrazee Apologies for jumping in here, I think @Rinechxn mentioned in the description that @GaraGuide had reviewed it?

> Co-Maintained By: @GaraGuide

Which I'm guessing is [this person on Bluesky](https://bsky.app/profile/gara-guide.bsky.social) who posts in Thai.

**Comment by Rinechxn** [comment]
@GaraGuide is No github account but have bluesky account is "@gara-guide.bsky.social"

**Comment by auroursa** [comment]
> @Rinechxn Thank you so much! @auroursa are you a Thai speaker? If so, was your review a signoff? If not, @Rinechxn  is there somebody we could recruit to act as a second pair of eyes on this?

I’m not a Thai speaker, but I believe the Thai community on Bluesky has already made improvements to it, as noted by @surfdude29.

The Hungarian translation was merged a few hours ago, so I think this PR will need some minor adjustments to resolve merge conflicts.

**Comment by pfrazee** [comment]
Wonderful! Thank you all so much

**Review by pfrazee — APPROVED** [review]

**Pull request merged**